source/slang/slang-check-modifier.cpp

```diff
@@ -1434,6 +1450,97 @@ bool isModifierAllowedOnDecl(bool isGLSLInput, ASTNodeType modifierType, Decl* d
     }
 }
 
+void GLSLBindingOffsetTracker::setBindingOffset(int binding, int64_t byteOffset)
+{
+    bindingToByteOffset.set(binding, byteOffset);
+}
+
+int64_t GLSLBindingOffsetTracker::getNextBindingOffset(int binding)
+{
+    int64_t currentOffset;
+    if (bindingToByteOffset.addIfNotExists(binding, 0))
+        currentOffset = 0;
+    else
+        currentOffset = bindingToByteOffset.getValue(binding) + sizeof(uint32_t);
+
+    bindingToByteOffset.set(binding, currentOffset + sizeof(uint32_t));
+    return currentOffset;
+}
+
+AttributeBase* SemanticsVisitor::checkGLSLLayoutAttribute(
+    UncheckedGLSLLayoutAttribute* uncheckedAttr,
+    ModifiableSyntaxNode* attrTarget)
+{
+    SLANG_ASSERT(uncheckedAttr->args.getCount() == 1);
+
+    Attribute* attr = nullptr;
+
+    // False if the current unchecked attribute node is deleted and does not result in a new checked
+    // attribute.
+    bool addNode = true;
+
+    const auto& name = uncheckedAttr->getKeywordName()->text;
+    if (name == "binding" || name == "set")
```

Comment:
Instead of using name comparisons, just dynamic cast `uncheckedAttr`:
```
if (auto bindingAttr = as<UncheckedGLSLBindingLayoutAttribute>(uncheckedAttr))
{
     ...
}
```

Reply:
Done for "binding" and "offset" but "set" and other glsl layout modifier names that will be added in the future do(or will) not have explicit types. Binding and offset explicitly have them just to work around the edge case for atomic counters.